### ISSUE TITLE
Undefined behaviour when interceptor method is overloaded in interceptor class PAYARA-1067

### ISSUE THREAD

**Issue by pdudits** [payara/Payara#1078]
We achieved very interesting behavior with following code constellation:

````
public class AuditInterceptor {
    @AroundInvoke
    public Object log(InvocationContext ctx) throws Exception {
        AuditLog auditLog = new AuditLog().....;
        log(auditLog);
        return ctx.proceed();
    }
    
    protected void log(Object obj) {
        logger.info("Audit: {}", obj);
    }
}
````

What happens is, that during deployment, the `InterceptorManager` will ask `LifecycleCallbackDescriptor` for a method reference to use as `@AroundInvoke` method. It will call `LifecycleCallback.getLifecycleCallbackMethodObject`, and it will return first method with name `log` in this class it finds. Which is, any of the two, and with JDK 8, the result is not stable, so some of the intercepted methods get correct interception, some of them broken one.

Since there is no constraint on method names in JSR-318, I believe `LifecycleCallbackDescriptor` needs to check the method annotations if  its `metadataSource == ANNOTATION`

**Comment by OndrejM** [comment]
Very interesting situation indeed :)
First I thought that interceptor method must be public, but it's not the case according to the Javadoc.

But the Javadoc of [AroundInvoke](https://docs.oracle.com/javaee/7/api/javax/interceptor/AroundInvoke.html) states that the method must consume a singe argument of type InvocationContext. Therefore it is even better to check if the method has this argument, to support both ANNOTATION and XML metadata.

**Comment by smillidge** [comment]
Closed by #1089
